.ci/check_code_formatting.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+VENV=build/venv-code-formatting
+CLANG_FORMAT_VERSION=13.0.0
+
+if [[ ! -d $VENV ]]; then
+    python3 -mvenv "$VENV"
```

Comment:
As opposed to `tox` as I once was (as it's a pain with custom C++ dependencies), I think that may actually simplify our linting setup.